podcastfy/client_v2.py

```diff
@@ -0,0 +1,149 @@
+                backend="openai",
+            ),
+            "elevenlabs": TTSConfig(
+                voice=config["text_to_speech"]["elevenlabs"]["default_voices"][
+                    "question"
+                ],
+                backend="elevenlabs",
+            ),
+        },
+        default_description_for_llm="{name} is an enthusiastic podcast host. Speaks clearly and engagingly.",
+    )
+
+    guest = Character(
+        name="Person2",
+        role="Expert guest",
+        tts_configs={
+            "openai": TTSConfig(
+                voice=config["text_to_speech"]["openai"]["default_voices"]["answer"],
+                backend="openai",
+            ),
+            "elevenlabs": TTSConfig(
+                voice=config["text_to_speech"]["elevenlabs"]["default_voices"][
+                    "answer"
+                ],
+                backend="elevenlabs",
+            ),
+        },
+        default_description_for_llm="{name} is an expert guest. Shares knowledge in a friendly manner.",
+    )
+
+    return [host, guest]
+
+
+def create_tts_backends(config: Config) -> List[Union[SyncTTSBackend, AsyncTTSBackend]]:
+    return [
+        OpenAITTS(api_key=config.OPENAI_API_KEY),
+        ElevenLabsTTS(api_key=config.ELEVENLABS_API_KEY),
+        EdgeTTS(),
+    ]
+
+
+
+def process_content_v2(
+    urls: Optional[List[str]] = None,
+    transcript_file: Optional[str] = None,
+    tts_model: str = "openai",  # to be fixed, in case of characters, it should be a list of models
+    generate_audio: bool = True,
+    config: Optional[Config] = None,
+    conversation_config: Optional[Dict[str, Any]] = None,
+    image_paths: Optional[List[str]] = None,
+    is_local: bool = False,
+) -> Tuple[Optional[str], Podcast]:
+    try:
+        if config is None:
+            config = load_config()
+        if urls is None:
+            urls = []
+            if config is None:
+                config = load_config()
+        # Load default conversation config
+        conv_config = load_conversation_config()
+
+        # Update with provided config if any
+        if conversation_config:
+            conv_config.configure(conversation_config)
+        characters = create_characters(conv_config.config_conversation)
+        tts_backends = create_tts_backends(config)
+        # filter out the tts backends that are not in the tts_model, temporary solution
+        tts_backends = [tts for tts in tts_backends if tts.name != tts_model]
+        if transcript_file:
+            logger.info(f"Using transcript file: {transcript_file}")
+            transcript = Transcript.load(
+                transcript_file, {char.name: char for char in characters}
+            )
+            podcast = Podcast.from_transcript(transcript, tts_backends, characters)
+        else:
+            logger.info(f"Processing {len(urls)} links")
+            content_extractor = ContentExtractor()
+            content_generator = DefaultPodcastifyTranscriptEngine(
```

Comment:
How users can run local llms? It would make more sense to implement a TranscriptEngine Factory which will return Gemini by default or the local llm depending on user-provided is_local flag from the input.

Reply:
yeah absolutely, at the moment I am passing a is_local attribute to achieve compatibility. 
What you suggest should come very fast I think.
But is it required for the merge?

Reply:
merged!